### ISSUE TITLE
Scripts don't load in iPad

### ISSUE THREAD

**Issue by mhardy** [quoid/userscripts#411]
The only script I've been able to successfully load is the one posted in this thread:

_Originally posted by @quoid in https://github.com/quoid/userscripts/issues/219#issuecomment-1092364644_

Oddly, I can't even load an identical copy of it with the name changed. I've tried dozens of times, restarted Safari, ran the Userscripts app numerous times re-selecting the directory, I simply can't get anything else to load.  I've tried a number of different .js scripts and .css examples, nothing else works.

I develop on a PC but have a Mac so I can get your whatever you need to help figure out the problem.  I *really* want this to work on my iPad... :)

**Comment by quoid** [comment]
Could you provide all your system details?

> I can't even load an identical copy of it with the name changed.

When you do not change the name, it runs as expected?

**Comment by mhardy** [comment]
> Could you provide all your system details?

iPad Pro (10.5-inch)
iOS: 15.6.1
128G avail/256G Capacity

> To verify, when you do not change the name, it runs as expected?

Getting anything to show up reliably is very difficult.  Even the one I got to work, takes a lot of fiddling before it shows up.  When I am able to get it to show up in the list, it does run as expected and shows the alert.

> Assuming the test user script loads, after you change the name, does it continue to load the previous user script?

I've only ever seen the one script show up in the list. I changed the name in the copy I made and it never showed up. Tonight however, I deleted everything from iCloud, the one that worked correctly disappeared from the list. Added the working script back to iCloud, but it now doesn't show up.  I've been refreshing for about 20 minutes, nothing.

Here's the one I got to work a few times.

`// ==UserScript==
// @name          Test JS Script
// @description   Test JS alert userscript for all sites except google.
// @match         <all_urls>
// @exclude-match *://*.google.com/*
// @noframes
// ==/UserScript==

alert("This is a test script");
`
![IMG_2144](https://user-images.githubusercontent.com/115857/211977494-ff3bf665-b9c7-4eb3-ba7d-005dbf4cce8e.PNG)

**Comment by quoid** [comment]
Thanks for the additional detail @mhardy - I think I might have an idea about what is going on. You didn't share details about *how* you are going about making these file updates, if you provide those I think that would give me a better idea.

However, below is a reply I made to a different user which I think might be applicable here. Please read it over and let me know if this is the cause of your issue: *"assuming changes to files in iCloud drive are immediate, when in reality they can take quite a bit of time"*.

> The issue you describe is actually quite common and I am have experienced it myself. I believe this is due to the way iCloud syncs file and do not believe this is an issue with the extension.

> In my experience, often when you change or upload a new file to iCloud it doesn’t immediately make it available on the device. For example, I add a new file/user script on my desktop but do not see it on my iOS device. Sometimes I have to wait quite some time before it shows up, but what I have found is that if I open the “Files app” on my iOS device I can see the file there but it has not yet been locally downloaded. I know this because I see a “cloud download” icon next to the file. To show you this, I attached an image to this email from my iPhone. 

> You can see in the image below that the files, “iOS Tests.js” and another file are “available” but not yet downloaded to the device by the presence of the “cloud download” icon. Most often, simply opening the Files app will trigger a sync and less often I have to actually click the icon that is shown.

> To summarize, after you’ve made changes on one device that you are expecting to show up on another, often you will need to open the Files app to trigger a sync or wait an indeterminate amount of time before it is available; it’s not instant with iCloud.

**Referenced Image:**

![IMG_8C6ECFD81383-1-1](https://user-images.githubusercontent.com/7660254/212093853-1ba70f7e-3acb-4b75-bc36-79068c8adf3c.jpeg)

**Comment by mhardy** [comment]
I think you cracked it!! I waited forever for files to sync assuming that was the issue but they never did.  I even opened the "Files app" many times assuming that would kick off the sync, it never did.  HOWEVER, I was in Icons mode instead of List mode so I never noticed the cloud icon.  I just switched to List mode and now I see that cloud Icon on all my files.  Click and hold the Cloud icon, select Download. Return to Safari, hit refresh, everything shows up and works!!!! Brilliant!!!  Great work @quoid.  Custom CSS & JS for sites on my iPad - game changer. Love it, thanks!
